### PR TITLE
feat: added carSalesCount data

### PR DESCRIPTION
**Pull request by GutierrezGo**
References https://www.figma.com/file/p8RIdxT0mcEQRf1g2dzpHa/2021---Q1---Reporting-%26-Billing?node-id=1207%3A295

Sales vs Capping + current transactions - https://car-sale-tracking-service.preprod.carforyou.ch/swagger-ui/index.html#/Car%20Sale/countCarSalesUsingGET.

## Motivation and context

Extend api-client for the widgets

## Thank you 🦀

### CODE REVIEW

File: src/services/reporting/carSales.ts [hunk 1]
```diff
@@ -98,3 +98,21 @@ export const deleteCarSalesRejection = async ({
...
+  status?: string
+  hasRejection?: boolean
+  options?: ApiCallOptions
+}): Promise<CarSaleCount> => {
+  const query = toQueryString({ status, hasRejection })
+  return fetchPath({
+    path: `dealers/${dealerId}/car-sales/count${query ? `?${query}` : ""}`,
+    options: { isAuthorizedRequest: true, ...options },
+  })
+}
```
Comment (Averethel) [inline src/services/reporting/carSales.ts:118]: How about returning a number directly?
```suggestion
}): Promise<number> => {
  const query = toQueryString({ status, hasRejection })
  const { count } =  fetchPath({
    path: `dealers/${dealerId}/car-sales/count${query ? `?${query}` : ""}`,
    options: { isAuthorizedRequest: true, ...options },
  })
  return count
}
```

File: src/types/models/carSales.ts [hunk 2]
```diff
@@ -1,5 +1,9 @@
 import { ApiSearchListing } from "../../types/models/listing"
 
+export interface CarSaleCount {
+  count: number
+}
+
```
Comment (Averethel) [inline src/types/models/carSales.ts:6]: This is not needed anymore, now that we switched to number.

**Review by Averethel — APPROVED** [review]

File: src/types/models/carSales.ts (hunk 2 above)
Comment (GutierrezGo) [reply src/types/models/carSales.ts:6]: True! thanks

**Pull request merged**